src/h2/utilities.py

```diff
@@ -603,6 +603,30 @@ def _combine_cookie_fields(headers, hdr_validation_flags):
         yield NeverIndexedHeaderTuple(b'cookie', cookie_val)
 
 
+def _split_outbound_cookie_fields(headers, hdr_validation_flags):
+    """
+    RFC 7540 § 8.1.2.5 allows for better compression efficiency,
+    to split the Cookie header field into separate header fields
+
+    We want to do it for outbound requests, as we are doing for
+    inbound
+    """
+    for header in headers:
+        if header[0] in (b'cookie', 'cookie'):
+            needle = b'; ' if isinstance(header[0], bytes) else '; '
+
+            if needle in header[1]:
+                for cookie_val in header[1].split(needle):
+                    if isinstance(header, HeaderTuple):
+                        yield header.__class__(header[0], cookie_val)
+                    else:
+                        yield header[0], cookie_val
```

Comment:
At this point, what are the trade-offs to simply always create `HeaderTuple`'s?

Reply:
Since we use the same approach in https://github.com/python-hyper/h2/blob/c7f967d77a6f32be374e33adca195e65a00afb63/src/h2/utilities.py#L549-L552, I don't mind either way.

Reply:
I also was thinking about the `NeverIndexedHeaderTuple` like
https://github.com/python-hyper/h2/blob/63b6b97e691334464a57f933bfd7ed444f364708/src/h2/utilities.py#L601-L603

as we know for sure that it's a `cookie` header

@Kriechi , do you have any suggestions about that?

Reply:
I'm not sure about that, but I see that all outbound cookies are handled using this pattern: `isinstance(header, HeaderTuple): ...` 

https://github.com/python-hyper/h2/blob/63b6b97e691334464a57f933bfd7ed444f364708/src/h2/utilities.py#L534-L538
https://github.com/python-hyper/h2/blob/63b6b97e691334464a57f933bfd7ed444f364708/src/h2/utilities.py#L548-L552

Also I'm not sure about using `NeverIndexedHeaderTuple` (e.g. how it can negatively affect compression)